src/enumo/scheduler.rs

```diff
@@ -99,4 +133,33 @@ impl Scheduler {
             }
         }
     }
+
+    pub fn run<L: SynthLanguage>(
+        &self,
+        egraph: &EGraph<L, SynthAnalysis>,
+        ruleset: &Ruleset<L>,
+    ) -> EGraph<L, SynthAnalysis> {
+        // Set timeout to something quite high (10 minutes) to reduce non-deterministic
+        // behavior. If egg terminates due to timeout, it's hard to reason about
+        // what did/didn't merge, and things vary a lot on different machines, so
+        // it is best to avoid terminating due to timeout.
+        self.run_internal(egraph, ruleset, None, Duration::from_secs(600))
```

Comment:
May need to up this for Halide / further parameterize it at some point, but that's a small change so let's just do this for now.